### PR TITLE
fix fit signature checking

### PR DESCRIPTION
**Pull request by niej**
- back port the fix to install wrong u-boot.bin.
- fix imx8mm and rpi3 missing u-boot defconfig items.

pico-imx7d, warp7, rpi3 and imx8mm fit signature is working with this PR.

### CODE REVIEW

**Comment by niej** [comment]
jenkins build will be pasted later.

File: meta-mbl-bsp-common/recipes-bsp/atf/atf.inc [hunk 1]
```diff
@@ -50,7 +50,7 @@ ATF_COMPILE_FLAGS = "-C ${S} \
       BL32=${DEPLOY_DIR_IMAGE}/optee/tee-header_v2.bin \
       BL32_EXTRA1=${DEPLOY_DIR_IMAGE}/optee/tee-pager_v2.bin \
       BL32_EXTRA2=${DEPLOY_DIR_IMAGE}/optee/tee-pageable_v2.bin \
-      BL33=${DEPLOY_DIR_IMAGE}/u-boot.bin \
+      BL33=${DEPLOY_DIR_IMAGE}/${UBOOT_IMAGE} \
```
Comment (diego-sueiro) [inline meta-mbl-bsp-common/recipes-bsp/atf/atf.inc]: Can you please add in the commit message an explanation about this change?

Comment (niej) [reply meta-mbl-bsp-common/recipes-bsp/atf/atf.inc]: OK. UBOOT_IMAGE is the right file in imx8mm case.

File: meta-mbl-distro/classes/mbl-uboot-sign.bbclass [hunk 2]
```diff
@@ -0,0 +1,61 @@
+# SPDX-License-Identifier: MIT
```
[the remark this answers is not in the document]
Comment (niej) [reply meta-mbl-distro/classes/mbl-uboot-sign.bbclass]: Already changed to bbappend.

File: meta-freescale-mbl/recipes-bsp/u-boot/u-boot-imx_%.bbappend [hunk 3]
```diff
@@ -6,13 +6,12 @@ inherit mbl-uboot-sign
 
 SRCBRANCH = "imx_v2018.03_4.14.78_1.0.0_ga-mbl"
 SRC_URI = "git://git.linaro.org/landing-teams/working/mbl/u-boot.git;protocol=https;nobranch=1 \
-	   file://0001-arm-imx-Add-mbl-specific-boot-option.patch \
```
[the remark this answers is not in the document]
Comment (niej) [reply meta-freescale-mbl/recipes-bsp/u-boot/u-boot-imx_%.bbappend]: removed.

**Comment by niej** [comment]
Below build is OK. But test report shows imx8mm and rpi3 does not have test result, though my local boot up test is OK.

http://jenkins.mbed-linux.arm.com/view/JunNie/job/jun-fix-fit-signature2/lastBuild/

http://e120856-lin.cambridge.arm.com/results.php?image_version=custom&image_number=9&custom_version=jun-fix-fit-signature2&lava_query=&lava_owner=&failures=false&submitter=junnie01&sort=by_job

**Comment by diego-sueiro** [comment]
> Below build is OK. But test report shows imx8mm and rpi3 does not have test result, though my local boot up test is OK.
> 
> http://jenkins.mbed-linux.arm.com/view/JunNie/job/jun-fix-fit-signature2/lastBuild/
> 
> http://e120856-lin.cambridge.arm.com/results.php?image_version=custom&image_number=9&custom_version=jun-fix-fit-signature2&lava_query=&lava_owner=&failures=false&submitter=junnie01&sort=by_job

The timeouts in the test farm are related to the flashing taking 14 minutes. But I can see that the boards are booting. Hence, I don't think it is related to this PR and I'll merge it.

**Pull request merged**